src/cats/BreedSelector.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useMemo } from "react";
+import clsx from "clsx";
+import { Status } from "./useAsync";
+import styles from "./BreedSelector.module.css";
+
+type BreedType = {
```

Comment:
nit: don't sufix types with `Type`. This can sometimes cause conflicts, and typically I rename the component instead of the type, since the type is used more.

Reply:
I guess if I should prefer to use interfaces over types, using the IFoo naming convention avoids having to come up with unique names for the interface and the function/component.

Reply:
I think it doesn't matter too much the convention you use, as long as you stick to it. Wherever you go will probably have something in place already.